### ISSUE TITLE
Supervisor keeps going “Unhealthy”

### ISSUE THREAD

**Issue by fredck** [home-assistant/supervisor#4381]
### Describe the issue you are experiencing

I'm constantly having the supervisor going unhealthy. The following error, more exactly:

<img width="612" alt="image" src="https://github.com/home-assistant/supervisor/assets/630060/3ee64d2d-3c1f-41da-9441-f766c02a83f4">

It points to the following help page, which offers a complicated solution for the problem:
https://www.home-assistant.io/more-info/unhealthy/privileged

This issue blocks me from updating anything else in ha.

## Workaround

After some research, I've found this talk:
https://community.home-assistant.io/t/supervisor-keeps-going-unheathly-dropping-out-from-being-privileged/279706

The solution proposed there is executing `sudo docker restart hassio_supervisor`

This works, which tells me that there is nothing wrong in my installation except the supervisor getting a bit lost.

This is a bit depressing. In my case, it involves opening a sh session in my server to run it. I need also to have this solution saved for reference somewhere so I remember what to do in such recurring situation.

Also, the help page makes the problem even more scary than it is.

### What type of installation are you running?

Home Assistant Supervised

### Which operating system are you running on?

Debian 11

### Steps to reproduce the issue

There is no specific way to reproduce it. It just happens... maybe after supervisor updates?

### Anything in the Supervisor logs that might be useful for us?

```txt
Just got a truncated error trace at the top of the logs with the following, which seems to be unrelated:


  File "/usr/src/supervisor/supervisor/api/middleware/security.py", line 268, in token_validation
    return await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/supervisor/supervisor/api/middleware/security.py", line 280, in core_proxy
    return await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/supervisor/supervisor/api/utils.py", line 62, in wrap_api
    answer = await method(api, *args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/supervisor/supervisor/api/store.py", line 178, in store_info
    ATTR_ADDONS: [
                 ^
  File "/usr/src/supervisor/supervisor/api/store.py", line 179, in <listcomp>
    self._generate_addon_information(self.sys_addons.store[addon])
  File "/usr/src/supervisor/supervisor/api/store.py", line 114, in _generate_addon_information
    ATTR_ADVANCED: addon.advanced,
                   ^^^^^^^^^^^^^^
  File "/usr/src/supervisor/supervisor/addons/model.py", line 227, in advanced
    return self.data[ATTR_ADVANCED]
           ^^^^^^^^^
  File "/usr/src/supervisor/supervisor/store/addon.py", line 19, in data
    return self.sys_store.data.addons[self.slug]
           ~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^
KeyError: 'a0d7b954_thelounge'
```
```


### System Health information

As described in the issue.

### Supervisor diagnostics

[config_entry-hassio-0f7faf0987dcd2404bcff3078d1bd5de.json.txt](https://github.com/home-assistant/supervisor/files/11749216/config_entry-hassio-0f7faf0987dcd2404bcff3078d1bd5de.json.txt)


### Additional information

_No response_

**Comment by fredck** [comment]
As requested, the system information:

---

## System Information

version | core-2023.6.1
-- | --
installation_type | Home Assistant Supervised
dev | false
hassio | true
docker | true
user | root
virtualenv | false
python_version | 3.11.3
os_name | Linux
os_version | 5.10.0-9-amd64
arch | x86_64
timezone | Europe/Warsaw
config_dir | /config

<details><summary>Home Assistant Community Store</summary>

GitHub API | ok
-- | --
GitHub Content | ok
GitHub Web | ok
GitHub API Calls Remaining | 4939
Installed Version | 1.32.1
Stage | running
Available Repositories | 1283
Downloaded Repositories | 9

</details>

<details><summary>Home Assistant Cloud</summary>

logged_in | false
-- | --
can_reach_cert_server | ok
can_reach_cloud_auth | ok
can_reach_cloud | ok

</details>

<details><summary>Home Assistant Supervisor</summary>

host_os | Debian GNU/Linux 11 (bullseye)
-- | --
update_channel | stable
supervisor_version | supervisor-2023.06.2
agent_version | 1.2.2
docker_version | 20.10.9
disk_total | 108.0 GB
disk_used | 12.7 GB
healthy | failed to load: Unhealthy
supported | true
supervisor_api | ok
version_api | ok
installed_addons | Terminal & SSH (9.7.1), Mosquitto broker (6.2.1), Zigbee2MQTT (1.31.2-1), Node-RED (14.2.2), Z-Wave JS UI (1.13.4), File editor (5.6.0), Matter Server (4.6.0), Studio Code Server (5.6.1)

</details>

<details><summary>Dashboards</summary>

dashboards | 6
-- | --
resources | 6
views | 6
mode | storage

</details>

<details><summary>Recorder</summary>

oldest_recorder_run | June 8, 2023 at 11:07
-- | --
current_recorder_run | June 9, 2023 at 13:36
estimated_db_size | 23.85 MiB
database_engine | sqlite
database_version | 3.41.2

</details>

**Comment by mdegat01** [comment]
This isn't a standard problem. This issue typically shows up when trying to run supervised in an unsupported way with other software interfering with docker (hence why the issue you linked is primarily ubuntu users). I have never seen this behavior in testing when the only things done on the host are the steps listed in the [supervised installer](https://github.com/home-assistant/supervised-installer) (the only supported way to run supervised).

Have you customized the host machine beyond just running the supervised installer? Any other software installed/running that might be interfering with docker or dependencies? The current list of supported/unsupported isn't an exhaustive list, we don't have the time to test everything that could possibly break supervised as the list is infinite. We just add checks as issues alert us to changes which caused problems.

> This is a bit depressing. In my case, it involves opening a sh session in my server to run it. I need also to have this solution saved for reference somewhere so I remember what to do in such recurring situation.

The expectation is that users running supervised are linux and docker experts, completely comfortable in the cli. Nearly all solutions proposed for supervised users are going to be cli solutions (starting with the install process itself) because sysdmins are effectively the target user there.

It sounds like you are not particularly comfortable in the CLI since you referred to restarting a docker container via the CLI as a complicated solution. In that case I might suggest using an HAOS type install instead of supervised. Since it sounds like you want an appliance install where everything can be managed in the UI (which is what HAOS is).

**Comment by mhoogenbosch** [comment]
I'm experiencing the exact same thing. It started acting up (as I try to remember it correctly) when I installed the beta of 2023.7, I believe I started using the beta from b3. 

hassio_supervisor = privileged:

```
mhoogenbosch@homeassistant ~                                                                                                                         [10:15:40]
> $ sudo docker inspect --format='{{.HostConfig.Privileged}}' hassio_supervisor
true
```
System info:

```
agent_version: 1.5.1
apparmor_version: 3.0.8
boot_timestamp: 1690215768633925
broadcast_llmnr: true
broadcast_mdns: true
chassis: desktop
cpe: ""
deployment: ""
disk_free: 331.6
disk_life_time: null
disk_total: 455.9
disk_used: 101.1
dt_synchronized: true
dt_utc: "2023-07-27T07:16:17.114638+00:00"
features:
- reboot
- shutdown
- services
- network
- hostname
- timedate
- os_agent
- resolved
- journal
- disk
- mount
hostname: homeassistant
kernel: 6.1.0-10-amd64
llmnr_hostname: homeassistant
operating_system: Debian GNU/Linux 12 (bookworm)
startup_time: 35.982921
timezone: Europe/Amsterdam
use_ntp: false
```

When I search the web, I often find people using Ubuntu which is not supported, or advising on doing a reboot of the host system. I have Debian 12 installed (which has become supported recently) and have tried rebooting multiple times.

Additional info:

```
> $ sudo ha supervisor info
addons:
- icon: true
  name: Samba share
  repository: core
  slug: core_samba
  state: started
  update_available: false
  version: 10.0.2
  version_latest: 10.0.2
- icon: true
  name: ESPHome
  repository: 5c53de3b
  slug: 5c53de3b_esphome
  state: started
  update_available: false
  version: 2023.7.0
  version_latest: 2023.7.0
- icon: true
  name: PS5 MQTT
  repository: df2164f9
  slug: df2164f9_ps5_mqtt
  state: started
  update_available: false
  version: 1.3.1
  version_latest: 1.3.1
- icon: true
  name: Log Viewer
  repository: a0d7b954
  slug: a0d7b954_logviewer
  state: started
  update_available: false
  version: 0.15.1
  version_latest: 0.15.1
- icon: true
  name: FTP
  repository: a0d7b954
  slug: a0d7b954_ftp
  state: started
  update_available: false
  version: 4.7.1
  version_latest: 4.7.1
- icon: true
  name: OneDrive Backup
  repository: de91e161
  slug: de91e161_hassio_onedrive_backup
  state: started
  update_available: false
  version: 2.1.2
  version_latest: 2.1.2
- icon: true
  name: Double Take Proxy
  repository: c7657554
  slug: c7657554_double-take-proxy
  state: started
  update_available: false
  version: 1.0.0
  version_latest: 1.0.0
- icon: true
  name: Exadel CompreFace
  repository: c7657554
  slug: c7657554_compreface
  state: started
  update_available: false
  version: 1.1.0
  version_latest: 1.1.0
- icon: true
  name: WireGuard
  repository: a0d7b954
  slug: a0d7b954_wireguard
  state: started
  update_available: false
  version: 0.8.1
  version_latest: 0.8.1
- icon: true
  name: Zigbee2MQTT Edge
  repository: 45df7312
  slug: 45df7312_zigbee2mqtt_edge
  state: started
  update_available: false
  version: edge
  version_latest: edge
- icon: true
  name: Frigate Proxy
  repository: ccab4aaf
  slug: ccab4aaf_frigate-proxy
  state: started
  update_available: false
  version: "1.3"
  version_latest: "1.3"
- icon: true
  name: File editor
  repository: core
  slug: core_configurator
  state: started
  update_available: false
  version: 5.6.0
  version_latest: 5.6.0
addons_repositories:
- name: Local add-ons
  slug: local
- name: Double Take Hass.io Add-ons
  slug: c7657554
- name: 'Home Assistant Add-on: Zigbee2MQTT'
  slug: 45df7312
- name: ESPHome
  slug: 5c53de3b
- name: Frigate hass.io addons
  slug: ccab4aaf
- name: PS5 MQTT Add-on
  slug: df2164f9
- name: Home Assistant Google Drive Backup Repository
  slug: cebe7a76
- name: Home Assistant Community Add-ons
  slug: a0d7b954
- name: Official add-ons
  slug: core
- name: Home Assistant Onedrive Backup Repository
  slug: de91e161
arch: amd64
auto_update: true
channel: beta
debug: false
debug_block: false
diagnostics: null
healthy: false
ip_address: 172.30.32.2
logging: info
supported: true
timezone: Europe/Amsterdam
update_available: false
version: 2023.07.2
version_latest: 2023.07.2
wait_boot: 5
```

```
> $ sudo ha core info
arch: amd64
audio_input: null
audio_output: null
boot: true
image: ghcr.io/home-assistant/qemux86-64-homeassistant
ip_address: 172.30.32.1
machine: qemux86-64
port: 8123
ssl: false
update_available: false
version: 2023.8.0b0
version_latest: 2023.8.0b0
watchdog: true
```
Mostly everything works, but when reloading automations for example, a timeout occures:

```
23-07-27 10:02:19 WARNING (MainThread) [supervisor.jobs] 'HomeAssistantCore.update' blocked from execution, system is not healthy - privileged
23-07-27 10:04:11 ERROR (MainThread) [supervisor.homeassistant.api] Error on call http://172.30.32.1:8123/api/core/state:

23-07-27 10:06:11 ERROR (MainThread) [supervisor.homeassistant.api] Error on call http://172.30.32.1:8123/api/core/state:
```
I don't have anything installed besides HA, no portainer or such a thing. It is a HP T540 thin client which sole purpose is to run HA. I installed debian supervised because the CPU wouldn't excied 1.5Ghz on HASSIO, but this was almost more than a year ago.

Also ran the supervised installer a couple of times, which seems to run fine:

```
> $ sudo dpkg -i homeassistant-supervised.deb
(Reading database ... 272063 files and directories currently installed.)
Preparing to unpack homeassistant-supervised.deb ...
[warn]
[warn] If you want more control over your own system, run
[warn] Home Assistant as a VM or run Home Assistant Core
[warn] via a Docker container.
[warn]
[warn] ModemManager service is enabled. This might cause issue when using serial devices.
Leaving 'diversion of /etc/NetworkManager/NetworkManager.conf to /etc/NetworkManager/NetworkManager.conf.real by homeassistant-supervised'
Leaving 'diversion of /etc/NetworkManager/system-connections/default to /etc/NetworkManager/system-connections/default.real by homeassistant-supervised'
Leaving 'diversion of /etc/docker/daemon.json to /etc/docker/daemon.json.real by homeassistant-supervised'
Leaving 'diversion of /etc/network/interfaces to /etc/network/interfaces.real by homeassistant-supervised'
Unpacking homeassistant-supervised (1.5.0) over (1.5.0) ...
Setting up homeassistant-supervised (1.5.0) ...
[info] Restarting NetworkManager
[info] Restarting docker service
PING checkonline.home-assistant.io (104.26.4.238) 56(84) bytes of data.
64 bytes from 104.26.4.238: icmp_seq=1 ttl=59 time=6.20 ms

--- checkonline.home-assistant.io ping statistics ---
1 packets transmitted, 1 received, 0% packet loss, time 0ms
rtt min/avg/max/mdev = 6.197/6.197/6.197/0.000 ms
[info] Install supervisor startup scripts
[info] Install AppArmor scripts
[info] Start Home Assistant Supervised
[info] Installing the 'ha' cli
[info] Within a few minutes you will be able to reach Home Assistant at:
[info] http://homeassistant.local:8123 or using the IP address of your
[info] machine: http://192.168.21.92:8123
```
Right after, it again is not healthy - privileged

**Comment by koemat** [comment]
I have the same problem on a Debian 12 machine. After updating the supervisor the installation is unhealthy. Only a reboot of the host machine resolves the problem.

**Comment by mhoogenbosch** [comment]
I solved my reloading issue. Was a custom integration called watchman, after removing this it is okay again. The supervisor stays unhealthy however

**Comment by mhoogenbosch** [comment]
After a reboot it shows healty btw, after removing watchman, by any chance you are running that integration too?

**Comment by mhoogenbosch** [comment]
After removing the watchman integration, my system stays healthy. On my behave, this thread can be closed.

**Comment by aIexus** [comment]
I have the same issue with unhealthy supervisor as not privileged...

Debian 12, reproduced adter update supervisor to 2023.08.3.

**Comment by marcgarciamarti** [comment]
> sudo docker restart hassio_supervisor



> ### Describe the issue you are experiencing
> I'm constantly having the supervisor going unhealthy. The following error, more exactly:
> 
> <img alt="image" width="612" src="https://user-images.githubusercontent.com/630060/245886645-3ee64d2d-3c1f-41da-9441-f766c02a83f4.png">
> It points to the following help page, which offers a complicated solution for the problem: https://www.home-assistant.io/more-info/unhealthy/privileged
> 
> This issue blocks me from updating anything else in ha.
> 
> ## Workaround
> After some research, I've found this talk: https://community.home-assistant.io/t/supervisor-keeps-going-unheathly-dropping-out-from-being-privileged/279706
> 
> The solution proposed there is executing `sudo docker restart hassio_supervisor`
> 
> This works, which tells me that there is nothing wrong in my installation except the supervisor getting a bit lost.
> 
> This is a bit depressing. In my case, it involves opening a sh session in my server to run it. I need also to have this solution saved for reference somewhere so I remember what to do in such recurring situation.
> 
> Also, the help page makes the problem even more scary than it is.
> 
> ### What type of installation are you running?
> Home Assistant Supervised
> 
> ### Which operating system are you running on?
> Debian 11
> 
> ### Steps to reproduce the issue
> There is no specific way to reproduce it. It just happens... maybe after supervisor updates?
> 
> ### Anything in the Supervisor logs that might be useful for us?
> ```
> Just got a truncated error trace at the top of the logs with the following, which seems to be unrelated:
> 
> 
>   File "/usr/src/supervisor/supervisor/api/middleware/security.py", line 268, in token_validation
>     return await handler(request)
>            ^^^^^^^^^^^^^^^^^^^^^^
>   File "/usr/src/supervisor/supervisor/api/middleware/security.py", line 280, in core_proxy
>     return await handler(request)
>            ^^^^^^^^^^^^^^^^^^^^^^
>   File "/usr/src/supervisor/supervisor/api/utils.py", line 62, in wrap_api
>     answer = await method(api, *args, **kwargs)
>              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/usr/src/supervisor/supervisor/api/store.py", line 178, in store_info
>     ATTR_ADDONS: [
>                  ^
>   File "/usr/src/supervisor/supervisor/api/store.py", line 179, in <listcomp>
>     self._generate_addon_information(self.sys_addons.store[addon])
>   File "/usr/src/supervisor/supervisor/api/store.py", line 114, in _generate_addon_information
>     ATTR_ADVANCED: addon.advanced,
>                    ^^^^^^^^^^^^^^
>   File "/usr/src/supervisor/supervisor/addons/model.py", line 227, in advanced
>     return self.data[ATTR_ADVANCED]
>            ^^^^^^^^^
>   File "/usr/src/supervisor/supervisor/store/addon.py", line 19, in data
>     return self.sys_store.data.addons[self.slug]
>            ~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^
> KeyError: 'a0d7b954_thelounge'
> ```
> 
>  ```
> 
> 
> ### System Health information
> 
> As described in the issue.
> 
> ### Supervisor diagnostics
> 
> [config_entry-hassio-0f7faf0987dcd2404bcff3078d1bd5de.json.txt](https://github.com/home-assistant/supervisor/files/11749216/config_entry-hassio-0f7faf0987dcd2404bcff3078d1bd5de.json.txt)
> 
> 
> ### Additional information
> 
> _No response_
> ```
This happens to me as well. I don't have nay third party software running on my server, but I do have HACS installed along with integrations installed through HACS. Rebooting the supervisor makes the warning disappear during weeks. If it was a persistent issue, should the message not appear way sooner?

thanks

**Comment by EinSchwerd** [comment]
Attached is my supervisor log file from the linked #4529 bug.
[supervisor.log](https://github.com/home-assistant/supervisor/files/12603221/supervisor.log)

According to the UI it occurred around 9AM on the 14th and there is one matching exception at 09:41:
`Sep 14 09:41:27 nuc hassio_supervisor[1369]: 23-09-14 09:41:27 ERROR (MainThread) [asyncio] Task exception was never retrieved
Sep 14 09:41:27 nuc hassio_supervisor[1369]: future: <Task finished name='Task-3615' coro=<Addon.watchdog_container() done, defined at /usr/src/supervisor/supervisor/addons/addon.py:1069> exception=AddonsJobError('Rate limit exceeded, more then 10 calls in 0:30:00')>
Sep 14 09:41:27 nuc hassio_supervisor[1369]: Traceback (most recent call last):
Sep 14 09:41:27 nuc hassio_supervisor[1369]:   File "/usr/src/supervisor/supervisor/addons/addon.py", line 1083, in watchdog_container
Sep 14 09:41:27 nuc hassio_supervisor[1369]:     await self._restart_after_problem(event.state)
Sep 14 09:41:27 nuc hassio_supervisor[1369]:   File "/usr/src/supervisor/supervisor/jobs/decorator.py", line 255, in wrapper
Sep 14 09:41:27 nuc hassio_supervisor[1369]:     raise on_condition(
Sep 14 09:41:27 nuc hassio_supervisor[1369]: supervisor.exceptions.AddonsJobError: Rate limit exceeded, more then 10 calls in 0:30:00`

**Comment by tricepsbrachii** [comment]
On Debian Bookworm, configured according to the instructions with no other modifications:
I get this lapse back into "unhealthy" whenever I get a notification for a new update to Hone Assistant Core. Restarting fixes the problem. But again, it only happens when I get a notification for an update to Home Assistant Core.

I also chatted with someone who has this problem too, and they think it's whenever `unattended-upgrades` gets kicked off. Not sure if they are related.

**Comment by strikeir13** [comment]
> On Debian Bookworm, configured according to the instructions with no other modifications: I get this lapse back into "unhealthy" whenever I get a notification for a new update to Hone Assistant Core. Restarting fixes the problem. But again, it only happens when I get a notification for an update to Home Assistant Core.
> 
> I also chatted with someone who has this problem too, and they think it's whenever `unattended-upgrades` gets kicked off. Not sure if they are related.

This also happens to me when a new update is available to Home Assistant Core. My system info is attached. Running HA Supervised on Debian 12 with no other Docker containers or other services.
[ha_sys_info.txt](https://github.com/home-assistant/supervisor/files/12807741/ha_sys_info.txt)

**Comment by ikifar2012** [comment]
I have been looking into this issue and am not entirely sure why this is happening, whenever the supervisor container is removed and recreated the supervisor spits out this error:
 ```log
INFO (MainThread) [supervisor.jobs] 'ResolutionFixup.run_autofix' blocked from execution, system is not healthy - privileged
```
I have noticed if the supervisor container is restarted again or the host is restarted it will clear itself up. Thus why you might get this error after a supervisor update and a reboot seems to fix it. The container that is being created is privileged just like on HAOS. Still working to find a solution if anyone has any insight as to what might be going on let me know

**Comment by alfs** [comment]
I'm having the same problem on Debian Bookworm which should be supported:

![Screenshot 2023-10-21 at 22 38 15](https://github.com/home-assistant/supervisor/assets/1869479/d43a53a6-a55b-4e18-a0fe-f78187ea8aaa)

 It happens with some weeks inbetween. Tracing the source code, all points to checking the container attributes. Both docker inspect and python container attributes list the container as privileged. Where else can I trace the "non-privilege"?


```
# docker inspect hassio_supervisor|grep Priv
            "Privileged": true,

# docker exec -ti hassio_supervisor python
Python 3.11.5 (main, Sep 28 2023, 17:20:50) [GCC 11.2.1 20220219] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import docker
>>> client = docker.from_env()
>>> container = client.containers.get('hassio_supervisor')
>>> container.attrs['HostConfig']['Privileged']
True
>>> 
```

Doing a ```docker restart hassio_supervisor``` solves the problem temporarily.

**Comment by esbenr** [comment]
Same issue here - it's so annoying.
Every time HA gets updated this occurs.

Workaround:
Restarting the host OS solves the problem until next supervisor update.
This is obviously a problem with HA and not the OS.

System:
Running HA supervised in docker on a Intel NUC with latest Debian OS.
According to specs this is supported.

With the frequent updates of HA, symptom treating this with manual restart of the host OS isn't satifactory.
Please get this fixed.

![image](https://github.com/home-assistant/supervisor/assets/560120/05a60c63-1e25-4d79-91e0-498551cc4377)

**Comment by Paul-Vdp** [comment]
I have been experiencing the same problem on two separate (supervised but supported) instances for quite some time now. The common pattern makes it pretty obvious how and where it all goes pearshaped - and hopefully for the devs what to do about it. After at least half a year of suffering from this bug, a solution would be welcomed by us 'supervised' users - of which there are many ...

**Comment by derailius** [comment]
just wanted to say that i'm experiencing the same issue

**Comment by elmigbot** [comment]
Same. The recent supervisor update prompted it. This only started happening after I updated to bookworm and used the latest HA pkg.

My fix is quite simple, although it goes against every instinct. **Click "ignore"**. After a few minutes, I can go about updating add-ons, etc, like nothing is wrong. (Because nothing is?)

**Comment by zagi988** [comment]
Having the same problem, on Bookworm. 
Worked great until I did the latest update. But now it says that this error happened "2 weeks ago" even tho it was never ignored. When it happened two weeks ago, I updated everything on the OS, re-run the ha-supervised installation and it went away. Until now.

**Comment by ch4d1** [comment]
I'm experiencing the same issue. It started for me back in early October on Debian 11. After a complete reinstall on Debian 12, the error message reappeared today. Restarting 'hassio_supervisor' resolves the problem temporarily.

**Comment by kshunterco** [comment]
Hopefully this is not piling on but...

I'm experiencing the same issue. Within the last 5 days I installed a fresh copy of Debian 12 on an RP4 and finished the supervised installation. Twice now it's simply said it's unhealthy, a reboot fixes everything.

**Comment by fireheadman** [comment]
Had it happen again.... but slightly different.
This time I noticed there was an update for supervisor.   I thought maybe this was the fix.    So I clicked to update it, then I popped up the issue again.

nevertheless, I had to reboot the host again to clear the error and this also seemed to release the alert/notice for the supervisor update, so maybe a fix was applied???  (time will tell).
![image](https://github.com/home-assistant/supervisor/assets/19847195/967b3f5f-480f-4ac6-b270-cdd51f9a0736)

After the reboot, I was able to apply the latest HA update without any issue.
![image](https://github.com/home-assistant/supervisor/assets/19847195/dd9f31d3-3ef3-43f1-baaf-3780464eafe8)

**Comment by wimb0** [comment]
I am having the same issue since last week.
Tried everything, reinstall convience script, os agent, supervisor update/reinstall.

I am running HA supervised on Debian 12 on a Dell Optiplex 3060 micro.
My Home Assistant installation has been running fine for over 3 year on this hardware.
(I updated from Debian 11 to Debian 12 about 2 months ago, and has been running fine since then)

Versions:
![image](https://github.com/home-assistant/supervisor/assets/11030068/489ac242-7d95-4f75-a5c7-63c7efb68e81)

**Comment by GSzabados** [comment]
It is not the answer for the mistery of this issue, but solves the problem quickly without restarting the whole host. (At least it does solve it for me.)

1. Going to Terminal

2. Typing `ha supervisor restart`

**Comment by robvangeloven** [comment]
Just to add my two cents, I have the same issues and it seems to occur around a new version notification. However, this time I have a new error (which would explain why running the installer script fixes things perhaps):

System is currently unhealthy because an attempt to update Supervisor to the latest version has failed. Use the link to learn more and how to fix this.

Perhaps it's a chicken and egg problem? Update needs the new supervisor to run, but the new supervisor cannot be updated because it needs the new update first

**Comment by weswitt** [comment]
seeing the same thing. fresh install using debian 12. host restart clears the issue, but unknown if it will return

**Comment by hansdej** [comment]
Same issue here on a Debian Bookworm system. (Installed with Buster but lived through 2 Debian stable upgrades successfully.)

**Comment by elmigbot** [comment]
This issue hasn't come back for me with the latest few supervisor updates. Perhaps it just hiccups once after updating to bookworm?

**Comment by ikifar2012** [comment]
> This issue hasn't come back for me with the latest few supervisor updates. Perhaps it just hiccups once after updating to bookworm?

It appears to only happen on Bookworm

**Comment by robvangeloven** [comment]
> This issue hasn't come back for me with the latest few supervisor updates. Perhaps it just hiccups once after updating to bookworm?

No, it hiccups a lot of times. Just randomly it seems, but always around new update availability (for me at least)

**Comment by ikifar2012** [comment]
> > This issue hasn't come back for me with the latest few supervisor updates. Perhaps it just hiccups once after updating to bookworm?
> 
> No, it hiccups a lot of times. Just randomly it seems, but always around new update availability (for me at least)

it seems to happen only when the container is recreated, it is privileged though

**Comment by wimb0** [comment]
Issue is still there after updating to Supervisor 2023.12.0.

**Comment by JohnnyM84** [comment]
Just upgraded to debian 12 and have the same issue with homeassistant-supervised (1.5.0), it's not going away with a reboot.
i installed and old version of the supervides installer version 1.3.1 and with that version for now it's healty.

**Comment by Lockszmith-GH** [comment]
In the meantime, a workaround for this is to restart the supervisor after it starts in an unhealthy state.

Run `sudo crontab -u root -e` to edit crontab as root:

Add the following line so that every 0, 23 or 46 of the hour check the logs and if supervisor started uhhealthy, restart the systemd service unit:

```cron
0/23 * * * * docker logs hassio_supervisor 2>&1 | grep -q "System is running in an unhealthy state and needs manual intervention!" && systemctl restart hassio-supervisor.service
```

While I hate these types of workaround, until somebody can explain **why** this is happening, this is the only recourse I have to have a healthy working system in the short term.

**Comment by weswitt** [comment]
is this being looked at by the dev team?

**Comment by whc2001** [comment]
> is this being looked at by the dev team?

Don't think so, since the managed HAOS is the only preferred installation method. Currently I am manually restarting the supervisoreach time and it works for me.

**Comment by okin** [comment]
I've been experiencing this same problem from time to time when there was an update for homeassistant I started over the web UI. Restarting the `hassio_supervisor` container or complete host will resolve the issue for me.

When the problem appears checking the container privileged status shows that it is privileged:
```
docker inspect -f '{{.HostConfig.Privileged }}' hassio_supervisor
true
```

The logs of the container always show a problem with privilege:
```
docker logs hassio_supervisor 2>&1 | grep udev
24-01-16 19:12:11 CRITICAL (MainThread) [supervisor.hardware.monitor] Not privileged to run udev monitor!
```

In the source [this](https://github.com/home-assistant/supervisor/blob/98c01fe1b39620dd9ad23bec63ed123cb9d88d5d/supervisor/hardware/monitor.py#L42) is the only place where the reason for an unhealthy system is the privileged state of a container.
We only get there if the setup of the monitor fails, but the [error does not contain additional information](https://github.com/pyudev/pyudev/blob/5d104b9c7192dff9a2f2ac54dd6418c7c198643e/src/pyudev/monitor.py#L119) as to the reason. (Side note: `EnvironmentError` was merged into `OSError` in Python 3.3)

As of now I don't know what leads to the problem to setup the `pyudev.Monitor`. I am not sure it is really a problem with missing privileged mode for the container.
I think a possible workaround in the code could be waiting for a few moments before trying to re-instantiate [the monitor](https://github.com/home-assistant/supervisor/blob/main/supervisor/hardware/monitor.py#L32).

**Comment by weswitt** [comment]
so maybe the message is just a false positive??

**Comment by ch4d1** [comment]
> > so maybe the message is just a false positive??
> As of now I can't say because I still lack the understanding why this happens.
> 
> The code we have today was mostly put in place as part of https://github.com/home-assistant/supervisor/pull/1566 to fix startup problems for the supervisor. The error seems to exist for quite some time.
> 
> While trying to create a simple test for the next time my installation shows the error I discovered that [setting the receive buffer size](https://github.com/pyudev/pyudev/blob/5d104b9c7192dff9a2f2ac54dd6418c7c198643e/src/pyudev/monitor.py#L258) could also result in `EnvironmentError`. This shines a light onto a reason for the message with the privileges: the calling process requires `CAP_NET_ADMIN` capabilities.
> 
> My test currently looks like this: `docker exec -it hassio_supervisor python -c 'from pyudev import Context, Monitor; c = Context(); print("setup context"); monitor = Monitor.from_netlink(c, "kernel"); print("setup monitor"); monitor.set_receive_buffer_size(32 * 1024 * 1024); print("setup buffer")'`
> Now I am waiting for my installation to be unhealthy again.

I think you can try to force it by downgrading supervisor to a previous version and then update it via the webui.

**Comment by timj-pdx** [comment]
I have been seeing same problem for about 6mo. Using all the tricks mentioned above, I have always been able to get hass restarted until now. Supervised install, Debian Bullseye. I am fully up-to-date on core/supervisor/Z-Wave JS as of yesterday. This morning I updated the OS. It was a minor update (3-4 packages) but one of the packages was *container*. I can't say exactly which packages were updated, but after that the supervisor fails to start.
Docker seems happy:
==> docker ps
CONTAINER ID   IMAGE                                                     COMMAND               CREATED             STATUS             PORTS                                   NAMES
bdded8b3122f   ghcr.io/home-assistant/amd64-hassio-observer:2023.06.0    "/usr/bin/observer"   3 seconds ago       Up 3 seconds       0.0.0.0:4357->80/tcp, :::4357->80/tcp   hassio_observer
a62e4eed970e   ghcr.io/home-assistant/amd64-hassio-multicast:2023.06.2   "/init"               About an hour ago   Up About an hour                                           hassio_multicast
0b681a70fc57   ghcr.io/home-assistant/amd64-hassio-audio:2023.12.0       "/init"               About an hour ago   Up About an hour                                           hassio_audio
10aeeebd818e   ghcr.io/home-assistant/amd64-hassio-dns:2023.06.2         "/init"               About an hour ago   Up About an hour                                           hassio_dns
9200f6e42964   ghcr.io/home-assistant/amd64-hassio-cli:2023.11.0         "/init"               About an hour ago   Up About an hour                                           hassio_cli
ff4a989ddb64   ghcr.io/home-assistant/amd64-hassio-supervisor:latest     "/init"               5 hours ago         Up About an hour                                           hassio_supervisor

### Tried This
==> docker restart hassio_supervisor
hassio_supervisor
==> echo $?
0
#### But it does not start HA
#### here is a snippet of the running 'docker logs hassio_supervisor'
24-01-19 12:51:56 INFO (SyncWorker_0) [supervisor.docker.manager] Cleaning hassio_dns application
24-01-19 12:51:56 ERROR (SyncWorker_0) [supervisor.docker.network] Can't link container to hassio-net: 500 Server Error for http+docker://localhost/v1.44/networks/05891387b0f9d002674bc531f627994ff9936b49d38645c0ad8f82ae16b9d0a6/connect: Internal Server Error ("invalid endpoint settings:
24-01-19 12:51:56 WARNING (SyncWorker_0) [supervisor.docker.manager] Can't attach hassio_dns to hassio-network!
24-01-19 12:51:57 INFO (MainThread) [supervisor.docker.dns] Starting DNS ghcr.io/home-assistant/amd64-hassio-dns with version 2023.06.2 - 172.30.32.3
24-01-19 12:51:57 INFO (MainThread) [supervisor.plugins.dns] Updated /etc/resolv.conf
24-01-19 12:51:57 INFO (MainThread) [supervisor.docker.interface] Attaching to ghcr.io/home-assistant/amd64-hassio-audio with version 2023.12.0
24-01-19 12:51:57 INFO (MainThread) [supervisor.plugins.audio] Starting Audio plugin
24-01-19 12:51:57 INFO (SyncWorker_0) [supervisor.docker.manager] Cleaning hassio_audio application
24-01-19 12:51:57 ERROR (SyncWorker_1) [supervisor.docker.network] Can't link container to hassio-net: 500 Server Error for http+docker://localhost/v1.44/networks/05891387b0f9d002674bc531f627994ff9936b49d38645c0ad8f82ae16b9d0a6/connect: Internal Server Error ("invalid endpoint settings:
24-01-19 12:51:57 WARNING (SyncWorker_1) [supervisor.docker.manager] Can't attach hassio_audio to hassio-network!

#### There is more but it is pretty clear there is no recovery here
#### not sure what how to proceed. I am willing to help in any way can. I am a pretty sophisticated Linux user. My hands-on experience with Docker is pretty limited.

**Comment by Lockszmith-GH** [comment]
@timj-pdx #349 seems to be related to what you are experiencing.

**Comment by dimkin-eu** [comment]
@timj-pdx docker 25 kicked in

```
sudo apt install \
docker-compose-plugin=2.21.0-1~debian.12~bookworm \
docker-ce-cli=5:24.0.7-1~debian.12~bookworm \
docker-buildx-plugin=0.11.2-1~debian.12~bookworm \
docker-ce=5:24.0.7-1~debian.12~bookworm \
docker-ce-rootless-extras=5:24.0.7-1~debian.12~bookworm
```

but this is not related to initial topic

**Comment by timj-pdx** [comment]
> I have been seeing same problem for about 6mo. Using all the tricks mentioned above, I have always been able to get hass restarted until now. Supervised install, Debian Bullseye. I am fully up-to-date on core/supervisor/Z-Wave JS as of yesterday. This morning I updated the OS. It was a minor update (3-4 packages) but one of the packages was _container_. I can't say exactly which packages were updated, but after that the supervisor fails to start. Docker seems happy: ==> docker ps CONTAINER ID IMAGE COMMAND CREATED STATUS PORTS NAMES bdded8b3122f ghcr.io/home-assistant/amd64-hassio-observer:2023.06.0 "/usr/bin/observer" 3 seconds ago Up 3 seconds 0.0.0.0:4357->80/tcp, :::4357->80/tcp hassio_observer a62e4eed970e ghcr.io/home-assistant/amd64-hassio-multicast:2023.06.2 "/init" About an hour ago Up About an hour hassio_multicast 0b681a70fc57 ghcr.io/home-assistant/amd64-hassio-audio:2023.12.0 "/init" About an hour ago Up About an hour hassio_audio 10aeeebd818e ghcr.io/home-assistant/amd64-hassio-dns:2023.06.2 "/init" About an hour ago Up About an hour hassio_dns 9200f6e42964 ghcr.io/home-assistant/amd64-hassio-cli:2023.11.0 "/init" About an hour ago Up About an hour hassio_cli ff4a989ddb64 ghcr.io/home-assistant/amd64-hassio-supervisor:latest "/init" 5 hours ago Up About an hour hassio_supervisor
> ### Tried This
> 
> ==> docker restart hassio_supervisor hassio_supervisor ==> echo $? 0
> #### But it does not start HA
> #### here is a snippet of the running 'docker logs hassio_supervisor'
> 
> 24-01-19 12:51:56 INFO (SyncWorker_0) [supervisor.docker.manager] Cleaning hassio_dns application 24-01-19 12:51:56 ERROR (SyncWorker_0) [supervisor.docker.network] Can't link container to hassio-net: 500 Server Error for http+docker://localhost/v1.44/networks/05891387b0f9d002674bc531f627994ff9936b49d38645c0ad8f82ae16b9d0a6/connect: Internal Server Error ("invalid endpoint settings: 24-01-19 12:51:56 WARNING (SyncWorker_0) [supervisor.docker.manager] Can't attach hassio_dns to hassio-network! 24-01-19 12:51:57 INFO (MainThread) [supervisor.docker.dns] Starting DNS ghcr.io/home-assistant/amd64-hassio-dns with version 2023.06.2 - 172.30.32.3 24-01-19 12:51:57 INFO (MainThread) [supervisor.plugins.dns] Updated /etc/resolv.conf 24-01-19 12:51:57 INFO (MainThread) [supervisor.docker.interface] Attaching to ghcr.io/home-assistant/amd64-hassio-audio with version 2023.12.0 24-01-19 12:51:57 INFO (MainThread) [supervisor.plugins.audio] Starting Audio plugin 24-01-19 12:51:57 INFO (SyncWorker_0) [supervisor.docker.manager] Cleaning hassio_audio application 24-01-19 12:51:57 ERROR (SyncWorker_1) [supervisor.docker.network] Can't link container to hassio-net: 500 Server Error for http+docker://localhost/v1.44/networks/05891387b0f9d002674bc531f627994ff9936b49d38645c0ad8f82ae16b9d0a6/connect: Internal Server Error ("invalid endpoint settings: 24-01-19 12:51:57 WARNING (SyncWorker_1) [supervisor.docker.manager] Can't attach hassio_audio to hassio-network!
> #### There is more but it is pretty clear there is no recovery here
> #### not sure what how to proceed. I am willing to help in any way can. I am a pretty sophisticated Linux user. My hands-on experience with Docker is pretty limited.

#### I grabbed some text from the 'apt' log that details which packages were updated right before I lost the ability to start hass.
reparing to unpack .../docker-buildx-plugin_0.12.1-1~debian.12~bookworm_amd64.deb ...
Preparing to unpack .../docker-ce-cli_5%3a25.0.0-1~debian.12~bookworm_amd64.deb ...
Preparing to unpack .../docker-ce_5%3a25.0.0-1~debian.12~bookworm_amd64.deb ...
Preparing to unpack .../docker-ce-rootless-extras_5%3a25.0.0-1~debian.12~bookworm_amd64.deb ...
Preparing to unpack .../docker-compose-plugin_2.24.1-1~debian.12~bookworm_amd64.deb ...

#### Yes this is Docker 25
#### The next step is rolling back these updates, not do any OS updates until HA is compatible with Docker 25
#### I agree this is not related to the initial topic. Sorry for the distraction

**Comment by timj-pdx** [comment]
@dimkin-eu Thank you so much, A quick cut&paste and  @everything came back to life. Should I submit a ticket to address problem with Docker 25? I am sure I am not the only one keeping his bookworm OS up-to-date.

**Comment by dimkin-eu** [comment]
@timj-pdx , its here https://github.com/home-assistant/supervisor/issues/4827

**Comment by donaldguy** [comment]
As I explored at much length in #4839 I am seeing the alarming situation (also on Debian bookworm) that:

1. the current hassio-supervisor apparmor [policy](https://github.com/home-assistant/version/blob/master/apparmor.txt) **DOES** always/consistently kill off the ability of the supervisor to initialize its udev monitor. Failing at the point of the` socket(PF_NETLINK, SOCK_RAW|SOCK_CLOEXEC|SOCK_NONBLOCK, NETLINK_KOBJECT_UEVENT)` syscall [past which I have not yet debugged or correlated; maybe `audit` will tell what needs knowing, maybe its into kernel] 
 
2. Perhaps more alarmingly, the reason I (and others) had only the impression that this permissioning "goes bad" later rather than simply "is" bad is because the policy **IS NOT** being applied to (any process in) the privileged container after a reboot, if that container already exists at boot - allowing it to work, but also potentially to do any number of things it shouldn't be allowed to.


As I see it, pending further investigation, this either:

- is a niche issue that will not effect HAOS because it is a regression in e.g. the kernel that has been (between 6.1.63 and 6.1.71) or will be fixed before OS takes the relevant package(s)
- is a deliberate or permanent change to how the permissions/capabilities are computed and is something HAOS could unwittingly upgrade into (with newer apparmor, systemd, docker, etc. than present)

**Comment by donaldguy** [comment]
Litmus test as to the (2) part of that:

when you have a healthy system and when you have an unhealthy system, try running

```shell
docker inspect hassio_supervisor --format='{{.State.Pid}}' | xargs ps Z --ppid
```

and see whether you are seeing approx

```
LABEL                               PID TTY      STAT   TIME COMMAND
unconfined                         2599 ?        S      0:00 s6-supervise s6-linux-init-shutdownd
unconfined                         2608 ?        S      0:00 s6-supervise s6rc-fdholder
unconfined                         2609 ?        S      0:00 s6-supervise s6rc-oneshot-runner
unconfined                         2712 ?        S      0:00 s6-supervise supervisor
unconfined                         2714 ?        S      0:00 s6-supervise watchdog
```

or

```
hassio-supervisor (enforce)      2599 ?        S      0:00 s6-supervise s6-linux-init-shutdownd
hassio-supervisor (enforce)      2608 ?        S      0:00 s6-supervise s6rc-fdholder
hassio-supervisor (enforce)      2609 ?        S      0:00 s6-supervise s6rc-oneshot-runner
hassio-supervisor (enforce)      2712 ?        S      0:00 s6-supervise supervisor
hassio-supervisor (enforce)      2714 ?        S      0:00 s6-supervise watchdog
```


`aa-status` or `apparmor_status` may be likewise instructive, but there its absence/presence rather than actual tagging

**Comment by spattinson** [comment]
+1 here, migrating to new hardware Debian bookworm on Rock 5b RK3588 with legacy rockchip kernel 5.10* so it has support for hardware video de/encoders and AI acceleration using rockchip NPU which Frigate uses. After wiping and reinstalling a few times I've finally got it all up and stable for now. Root cause seems to be inside alpine use of udev according to analysis by @donaldguy in https://github.com/home-assistant/supervisor/issues/4839
Is one solution to use Debian Bullseye instead? Everyone here reports using Bookworm from what I can see

**Comment by avandorp** [comment]
> ```shell
> docker inspect hassio_supervisor --format='{{.State.Pid}}' | xargs ps Z --ppid
> ```
gives "unconfined" for me in the healthy state.

**Comment by ikac2000** [comment]
> In the meantime, a workaround for this is to restart the supervisor after it starts in an unhealthy state.
> 
> Run `sudo crontab -u root -e` to edit crontab as root:
> 
> Add the following line so that every 0, 23 or 46 of the hour check the logs and if supervisor started uhhealthy, restart the systemd service unit:
> 
> ```
> 0/23 * * * * docker logs hassio_supervisor 2>&1 | grep -q "System is running in an unhealthy state and needs manual intervention!" && systemctl restart hassio-supervisor.service
> ```
> 
> While I hate these types of workaround, until somebody can explain **why** this is happening, this is the only recourse I have to have a healthy working system in the short term.

Just to ask, having this in crontab - once the error happens it stays in the log, so condition will be fullfilled, and every 23 minutes supervisor will be restarted from that moment?

I have it in the log, restarted supervisor, and it is still in the logs history.

**Comment by Lockszmith-GH** [comment]
@ikac2000 I found that the message shows only once for me in the logs, at the beginning of the superviser log, after a restart.

so yes, if the restart doesn't remove the message, something else is going on.
For me, the service restart resovles the issue.

It seems that for you that is not the case.

Have you confirmed that after `systemctl restart hassio-supervisor.service` the log rotates?

**Comment by ikac2000** [comment]
> @ikac2000 I found that the message shows only once for me in the logs, at the beginning of the superviser log, after a restart.
> 
> so yes, if the restart doesn't remove the message for you, something else is going on. For me, the service restart resovles the issue.
> 
> It seems that for you that is not the case.
> 
> Have you confirmed that after `systemctl restart hassio-supervisor.service` the log rotates?

Thanks for the answer, but unfortunatelly in my case (RPI 5 on Raspbian) log stays, and gets enriched with restarted lines.

Is there a setting to rotate the log after restart, or something similar maybe (I am not so good in this topics yet)?

But adding -- since 24m for sure resolves auto action issue.

Still waiting for the problem in general to be resolved.

**Comment by nyok92** [comment]
Same issue here with debian bookworm on rock5b with rockchip linux 5.10
Solved with docker restart hassio supervisor container

**Comment by shaatie** [comment]
I am facing the same issue. I am running Home Assistant Supervised on bookworm on Rasp Pi 4 
Restarting the supervisor container as suggested got rid of the warning and also allowed me to run the HA update.

**Comment by okin** [comment]
I had some time today to look around at a system with an _unhealthy_ state.

# state: unhealthy

```
# docker exec -it hassio_supervisor python -c 'from pyudev import Context, Monitor; c = Context(); print("setup context"); monitor = Monitor.from_netlink(c, "kernel"); print("setup monitor"); monitor.set_receive_buffer_size(32 * 1024 * 1024); print("setup buffer")'
setup context
setup monitor
setup buffer
```

With this all working I do not think any problem lies here.

Going into the apparmor direction that @donaldguy brought up seemed more promising after this.

```
# docker inspect hassio_supervisor --format='{{.State.Pid}}' | xargs ps Z --ppid
LABEL                               PID TTY      STAT   TIME COMMAND
hassio-supervisor (enforce)       30957 ?        S      0:00 s6-supervise s6-linux-init-shutdownd
hassio-supervisor (enforce)       30966 ?        S      0:00 s6-supervise s6rc-oneshot-runner
hassio-supervisor (enforce)       30967 ?        S      0:00 s6-supervise s6rc-fdholder
hassio-supervisor (enforce)       31015 ?        S      0:00 s6-supervise supervisor
hassio-supervisor (enforce)       31017 ?        S      0:00 s6-supervise watchdog
hassio-supervisor (enforce)       31093 ?        Sl     0:00 /usr/local/bin/python3 /usr/local/lib/python3.12/site-packages/debugpy/adapter --for-server 60223 --host 0.0.0.0 --port 33333 --server-access-token <removed>
```

# after container restart

Restarting the container brings up a healthy system state again.

```
# docker inspect hassio_supervisor --format='{{.State.Pid}}' | xargs ps Z --ppid
LABEL                               PID TTY      STAT   TIME COMMAND
unconfined                        57136 ?        S      0:00 s6-supervise s6-linux-init-shutdownd
unconfined                        57145 ?        S      0:00 s6-supervise s6rc-oneshot-runner
unconfined                        57146 ?        S      0:00 s6-supervise s6rc-fdholder
unconfined                        57194 ?        S      0:00 s6-supervise supervisor
unconfined                        57196 ?        S      0:00 s6-supervise watchdog
unconfined                        57254 ?        Sl     0:00 /usr/local/bin/python3 /usr/local/lib/python3.12/site-packages/debugpy/adapter --for-server 37439 --host 0.0.0.0 --port 33333 --server-access-token <removed>
```

Going after `aa-status` I find that that the processes being in _enforce mode_ all revolve around _docker_ and _hassio-supervisor_.

I did a bit of poking around with `systemd-analyze` but the for me it looks like the ordering of units isn't a problem here - first `apparmor.service` gets started, then `hassio-apparmor.service` (which should load the profiles) and after `docker.service` our `hassio-supervisor.service` is started.
I'd say this is as expected and at least to me nothing did stick out that pointed to a problem during boot.

A `docker inspect -f '{{ .HostConfig.SecurityOpt }}' hassio_supervisor` shows me an `[apparmor=hassio-supervisor label=disable]`.
A `docker inspect -f '{{.AppArmorProfile }}' hassio_supervisor` will show an `unconfined`.
From the messages seen through `dmesg | grep audit` I can only find messages that show that the `unconfined` profile gets applied to the supervisor.

With `unconfined` being the working state I wonder how the `enforced` state gets applied to the processes and what would be the correct thing here (having them _unconfined_ seems unusual to me, because we have a _hassio-supervisor_ label).

**Comment by rotary65** [comment]
Nice work okin. This seems to highlight a need for clarity on how security policies are enforced and managed within the system.

**Comment by nirkons** [comment]
Just wanted to chime in and say that I've been experiencing this issue for years, though previously I was on ubuntu, I migrated everything to Debian just to resolve this issue, but it chased me even to the new installation. No other docker containers except those created by homeassistant are running

**Comment by bluearcher-bc** [comment]
Been dealing with this for ages, with the work of several people in the last couple months it does seem like there is an issue to be addressed here, and not just non-standard installs or other user errors. @mdegat01 you seem to be at least one person that's commented on this, can we get this triaged using the latest information in this thread from @okin and @donaldguy ? Does Nabu Casa agree there is an issue or not?

Like others, I also have this issue, almost every single time I try to update HA since upgrading to Bookworm, on a box that is well maintained, regularly updated, and not running any other software.

Currently in a bad state:

```
brad@hass:~$ sudo docker inspect hassio_supervisor --format='{{.State.Pid}}' | xargs ps Z --ppid
LABEL                               PID TTY      STAT   TIME COMMAND
hassio-supervisor (enforce)     4075542 ?        S      0:00 s6-supervise s6-linux-init-shutdownd
hassio-supervisor (enforce)     4075551 ?        S      0:00 s6-supervise s6rc-fdholder
hassio-supervisor (enforce)     4075552 ?        S      0:00 s6-supervise s6rc-oneshot-runner
hassio-supervisor (enforce)     4075600 ?        S      0:00 s6-supervise supervisor
hassio-supervisor (enforce)     4075602 ?        S      0:00 s6-supervise watchdog
brad@hass:~$ uname -a
Linux hass 6.1.0-18-amd64 #1 SMP PREEMPT_DYNAMIC Debian 6.1.76-1 (2024-02-01) x86_64 GNU/Linux
```

**Comment by whc2001** [comment]
> > In the meantime, a workaround for this is to restart the supervisor after it starts in an unhealthy state.
> > Run `sudo crontab -u root -e` to edit crontab as root:
> > Add the following line so that every 0, 23 or 46 of the hour check the logs and if supervisor started uhhealthy, restart the systemd service unit:
> > ```
> > 0/23 * * * * docker logs hassio_supervisor 2>&1 | grep -q "System is running in an unhealthy state and needs manual intervention!" && systemctl restart hassio-supervisor.service
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > While I hate these types of workaround, until somebody can explain **why** this is happening, this is the only recourse I have to have a healthy working system in the short term.
> 
> Just to ask, having this in crontab - once the error happens it stays in the log, so condition will be fullfilled, and every 23 minutes supervisor will be restarted from that moment?
> 
> I have it in the log, restarted supervisor, and it is still in the logs history.
> 
> Should there be a --since 24m or similar, to limit log search, like:
> 
> 0/23 * * * * docker logs hassio_supervisor --since 24m 2>&1 | grep -q "System is running in an unhealthy state and needs manual intervention!" && systemctl restart hassio-supervisor.service

I quickly realized this issue by changing `minute` to `*/5` and found out my supervisor is being restarted vigorously. After some digging I found out that the output of `ha supervisor info` command will have a field indicating healthy or not. Here is what I am currently using:
```bash
#!/bin/bash
while true; do
    if ha supervisor info 2>&1 | grep -q "healthy: false"; then
        echo "Unhealthy detected, restarting" | systemd-cat -t $(basename "$0") -p emerg
        systemctl restart hassio-supervisor.service
        sleep 600
    else
        sleep 5
    fi
done
```
Got this snippet running on boot (I am using systemd) and it should be all good. Haven't observed any unhealthy recently though...

**Comment by Lockszmith-GH** [comment]
@nirkons 
> Just wanted to chime in ....
Did you just repeast [this](https://github.com/home-assistant/supervisor/issues/4381#issuecomment-1871479830)?

> ```shell
> if ha supervisor info 2>&1 | grep -q "healthy: false"; then
> ```
@whc2001 this is great!
I really wish they'd [make it available](https://github.com/home-assistant/supervised-installer/issues/342) from within HA, but with your finding I think I can find a way to make it appear as a sensor.

**Comment by ikac2000** [comment]
As this looks as a good solution, can someone just tell me how to implement this script - I guess I will need to create a .sh file, but where to put it and how to start it?

many thanks!

**Comment by whc2001** [comment]
> > ```shell
> > if ha supervisor info 2>&1 | grep -q "healthy: false"; then
> > ```
> 
> @whc2001 [this](https://github.com/home-assistant/supervisor/issues/4381#issuecomment-2027888383) is great! I really wish they'd [make it available](https://github.com/home-assistant/supervised-installer/issues/342) from within HA, but with your finding I think I can find a way to make it appear as a sensor.
> 
> @nirkons
> 
> > Just wanted to chime in ....
> 
> [Did you just](https://github.com/home-assistant/supervisor/issues/4381#issuecomment-2026811229) repeat [this](https://github.com/home-assistant/supervisor/issues/4381#issuecomment-1871479830)? If so, make sure you understand the caveats. looks like @whc2001's solution is a safer one.

I initially thought you can use command line sensor for this, but then I realized with HAOS/Supervised installation, the command is executed inside the container where it cannot reach `/usr/bin/ha` installed on the host OS. You might need to take this into account.




> As this looks as a good solution, can someone just tell me how to implement this script - I guess I will need to create a .sh file, but where to put it and how to start it?
> 
> many thanks!

You can create a `.sh` script anywhere (I just dropped it inside my home directory), paste the script into it, `chmod +x` on it and write a systemd script for it to auto start. Here is mine (/etc/systemd/system/supervisor-fix.service):
```ini
[Unit]
Description=Supervisor Unhealthy Fix

[Service]
ExecStart=/home/hass/supervisor_fix.sh

[Install]
WantedBy=multi-user.target
```
Change the `ExecStart` to your script location, save, `systemctl enable supervisor-fix`, `systemctl start supervisor-fix` and `systemctl status supervisor-fix` to check if it's running.

**Comment by donaldguy** [comment]
#4839 is definitely about to get staled out 

So I wanted to pull over this point about the root cause, in the current apparmor

> specifically, as I sort of said, believe the issue with the policy is, 
> 
> 
> 
> for some reason now-and-not-before, the interpretation of `deny network raw` 
> 
> https://github.com/home-assistant/version/blob/master/apparmor_stable.txt#L8 
> 
> 
> 
> now applying also to `netlink` domain, meaning the 
> 
> 
> 
> I don't believe just adding an `allow network netlink raw` will work ... (but maybe it will with the right ordering?)
> 
> 
> 
> I think giving it a cut out will require instead saying like... 
> 
> 
> 
> ```
> 
> deny network inet raw,
> 
> deny network ax25 raw,
> 
> deny network ipx raw,
> 
> deny network appletalk raw,
> 
> deny network netrom raw,
> 
> deny network bridge raw,
> 
> deny network atpmvc raw,
> 
> deny network x25 raw,
> 
> deny network inet6 raw,
> 
> deny network rose raw,
> 
> deny network netbui raw,
> 
> deny network security raw,
> 
> deny network key raw,
> 
> deny network packet raw,
> 
> deny network ash raw,
> 
> deny network econet raw,
> 
> deny network atmsvc raw,
> 
> deny network sna raw,
> 
> deny network irda raw,
> 
> deny network ppox raw,
> 
> deny network wanpipe raw,
> 
> deny network bluetooth raw,
> 
> deny network unix raw,
> 
> deny network rds raw,
> 
> deny network llc raw,
> 
> deny network can raw,
> 
> deny network tipc raw,
> 
> deny network iucy raw,
> 
> deny network rxrpc raw,
> 
> deny network isdn raw,
> 
> deny network phonet raw,
> 
> deny network ieee802154 raw,
> 
> deny network caif raw,
> 
> deny network alg raw,
> 
> deny network nfc raw,
> 
> deny network vsock raw,
> 
> deny network mpls raw,
> 
> deny network ib raw,
> 
> ```
> 
> 
> 
> 
> 
> Of course ... the logical question there is ... does it actually need non-raw access to most of those domains? (probs no)
> 
> 
> 
> so probs the sweeping [allow] `network` [*] [*] [*] of line 7 probs should go
> 
> 
> 
> maybe want like(?)
> 
> 
> 
> ```
> 
> allow network inet tcp,
> 
> allow network inet6 tcp,
> 
> allow network bluetooth,
> 
> allow network netlink,
> 
> ```
> 
> 
> 
> (maybe bridge? unix?)
> 
> 
> 
> 


A step above that is an issue in moby/docker I have more specifically identified [down to lines of code) and raised to their security team back on February 22nd 

- they were somewhat unclear whether they consider it a CVE given that it constitutes inconsistent security posture related to a situations being … sometimes more secure than they ever intended (but have left as is a long time)

I asked for a follow up back on March 17th but got no reply (so have heard nothing since 2/27)

[attempting to do a "responsible disclosure" there is why I've got quite in these discussions]

**Comment by donaldguy** [comment]
but in the meantime, if anyone wants a durable workaround (that would make things less secure in theory but not practice), you could just replace

https://github.com/home-assistant/supervised-installer/blob/c787aa7bc03c65529f3b3ac433150ce0967c3d66/homeassistant-supervised/usr/sbin/hassio-supervisor#L90

With

```
--privileged \
```

in `/usr/sbin/hassio-supervisor` (or `/usr/sbin/hasos-supervisor`)

Honestly I don't know why I haven't done that on my own machine

**Comment by JCalvi** [comment]
Hopefully the fix included in the above pull request stops this as I have the exact same issue and use the following script running every 15 minutes from Cron.

If it doesn't, posted below to help others and to add my voice to this issue. Happens every supervisor update.


Mine had a no route error so healthy: fail would not be returned either. I check NOT healthy true instead.

Note: I also check the core as i had a weird instance of that crashing while the rest of my debian 12 was running fine and ha core start fixed it quickly, otherwise a docker restart will fix most issues. I do this by actually testing I can reach the web port.

```
#!/bin/bash

# Script checks if Home Assistant is responsive on port and attempts 2 level restart if not.
# Level 1 restarts core first, if this is not enough Level 2 restarts all docker containers.
# J. Calvi 2024

HOST_NAME="192.168.1.2"
HOST_PORT="8123"

#Check Supervisor and restart if required.
if ha supervisor info | grep "healthy: true" > /dev/null; then
  echo -e "$(date): Home Assistant Supervisor Running"
else
  echo -e "$(date): Home Assistant Supervisor Not Running"
  #systemctl restart hassio-supervisor
fi

#Check Core and 2 Level restart if required.
for i in {1..2};
do
	if ( (exec 3<>/dev/tcp/${HOST_NAME}/${HOST_PORT}) 2> /dev/null); then
	  echo -e "$(date): Home Assistant Core Running"
	  exit;
	else
	  if (($i == 1)); then
	    echo -e "$(date): Home Assistant Core not responding, restarting core"
	    ha core start
	  else
	  	echo -e "$(date): Home Assistant Core not restarting, restarting docker"
		docker restart $(docker ps -a -q)
	  fi
	fi
done

exit;
```

**Comment by deiger** [comment]
Should this be considered fixed before [apparmor_stable.txt](https://github.com/home-assistant/version/blob/master/apparmor_stable.txt) is updated as well?

**Comment by agners** [comment]
The closing is done automatically when GitHub sees the keyword "fixed" along with an issue number. This is quite typical, issues usually get closed when the PR is merged to the main branch. Issue tracker are really a tool for developers, but yeah for users it gets a bit confusing :cry: Also, the version repo is a bit special in that sense as we have different channels in the same repo. :shrug:

**Comment by deiger** [comment]
Right, I'm aware of the auto-closing logic of GitHub.
It was really more of a process question, as I have not seen an automated process for maturing the apparmor configuration to stable.
